### DISCUSSION TITLE
Handling a single schema.sql

### DISCUSSION THREAD

**Discussion by erickhouse** [ariga/atlas#2134] [Q&A]
Hi. 

The atlas API expects a single schema.sql file when generating migrations (`atlas migrate diff`) . How are people developing against this API restriction? 

My assumption is that there's an expected pre-processing step to generate a single schema.sql file. What tools are people using to handle this step?

**Comment by a8m** [comment]
Hey @erickhouse 👋 

The desired state can be defined as a schema directory, which is a list of files executed in lexicographical order. The only restriction is that it must contain an atlas.sum file; hence, the directory will be managed as a migration directory. Please give this a try and let me know if you need any help with it.

```shell
➜  example echo 'create table t1(c int);' > schema/1.sql

➜  example echo 'create table t2(c int);' > schema/2.sql

➜  example atlas migrate hash --dir file://schema

➜  example atlas schema apply \
--to file://schema \
--dev-url "sqlite://dev?mode=memory" \
--url sqlite://file.db
-- Planned Changes:
-- Create "t1" table
CREATE TABLE `t1` (`c` int NULL);
-- Create "t2" table
CREATE TABLE `t2` (`c` int NULL);
✔ Apply
```
